src/cursor/mod.rs

```diff
@@ -383,7 +366,7 @@ impl CursorData {
         Ok(())
     }
 
-    fn fill_buf_impl<'a>(&self, list: &'a BufList) -> &'a [u8] {
```

Comment:
Don't think this code change is required.

Reply:
```
error: lifetime may not live long enough
   --> src/cursor/mod.rs:372:41
    |
369 |     fn fill_buf_impl(&self, list: &BufList) -> &[u8] {
    |                      -            - let's call the lifetime of this reference `'1`
    |                      |
    |                      let's call the lifetime of this reference `'2`
...
372 |             Some((chunk, chunk_pos)) => &chunk.as_ref()[chunk_pos..],
    |                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ associated function was supposed to return data with lifetime `'2` but it is returning data with lifetime `'1`
    |
help: consider introducing a named lifetime parameter and update trait if needed
    |
369 |     fn fill_buf_impl<'a>(&'a self, list: &'a BufList) -> &[u8] {
    |                     ++++  ++              ++
```

Reply:
this one is required, sadly (not sure why, it is weird)

Reply:
nvm yeah that is required.